app/src/store/modules/explorer/curation/actions.js

```diff
@@ -88,5 +89,61 @@ export default {
 
     const response = await fetchResponse.json()
     return { response, identifier: chartInstanceObject.identifier }
+  },
+
+  async lookupOrcid ({ commit }, orcidId) {
+    const unhyphenated = /^\d{15}(\d|X)$/.test(orcidId)
+    unhyphenated && (orcidId = orcidId.replace(/^\(?(\d{4})\)?(\d{4})?(\d{4})?(\d{3}(\d|X))$/, '$1-$2-$3-$4'))
+
+    if (isValidOrcid(orcidId)) {
+      // TODO: update the endpoint route name
+      const url = `/api/knowledge/images?uri=http://orcid.org/${orcidId}&view=describe`
+      const response = await fetch(url, {
+        method: 'GET'
+      })
+      if (!response || response?.statusText !== 'OK') {
+        const snackbar = {
+          message: response.message || 'Something went wrong while fetching orcid data',
+          duration: 5000
+        }
+        return commit('setSnackbar', snackbar, { root: true })
+      }
+
+      if (response.status === 201) {
+        return
+      }
+      const responseData = await response.json()
+      const cpResult = responseData.filter(entry => entry['@id'] === `http://orcid.org/${orcidId}`)
+      if (cpResult.length) {
+        return commit('setOrcidData', cpResult[0])
+      } else {
+        // No results were returned
+        return commit('setOrcidData', 'invalid')
+      }
+    } else {
+      // Incorrect format
+      return commit('setOrcidData', 'invalid')
+    }
+  },
+
+  async lookupDoi ({ commit }, inputDoi) {
+    const url = `/api/curate/getdoi/${inputDoi}`
+    const response = await fetch(url, {
+      method: 'GET'
+    })
+    if (!response || response?.statusText !== 'OK') {
+      if (!response || response?.statusText !== 'OK') {
```

Comment:
I made a mistake initially with this too. We may not need the first check i.e. `!response` if we are already optionally chaining.